lib/segment/src/index/field_index/numeric_index/immutable_numeric_index.rs

```diff
@@ -151,13 +150,19 @@ impl<T: Encodable + Numericable> NumericKeySortedVec<T> {
         }
     }
 
-    fn find_end_index(&self, bound: Bound<NumericIndexKey<T>>) -> usize {
+    /// Find the end index of the range.
+    ///
+    /// Scans only starting from `start` to avoid repeated work.
+    fn find_end_index(&self, start: usize, bound: Bound<NumericIndexKey<T>>) -> usize {
```

Comment:
Instead of a documentation comment, I propose to add asserts like `assert!(start <= self.data.len() && start <= bound.start)`